tox.ini

```diff
@@ -156,21 +151,20 @@ envlist =
     python-mlmodel_openai-openailatest-{py37,py38,py39,py310,py311,py312},
     python-mlmodel_sklearn-{py37}-scikitlearn0101,
     python-mlmodel_sklearn-{py38,py39,py310,py311,py312}-scikitlearnlatest,
-    python-template_genshi-{py27,py37,py38,py39,py310,py311,py312}-genshilatest,
+    python-template_genshi-{py37,py38,py39,py310,py311,py312}-genshilatest,
     python-template_jinja2-{py38,py39,py310,py311,py312}-jinja2latest,
     python-template_jinja2-py37-jinja2030103,
-    python-template_mako-{py27,py37,py38,py39,py310,py311,py312},
+    python-template_mako-{py37,py38,py39,py310,py311,py312},
     rabbitmq-messagebroker_pika-{py37,py38,py39,py310,py311,py312,pypy310}-pikalatest,
     redis-datastore_redis-{py37,py38,py39,py310,py311,py312,pypy310}-redis{0400,latest},
     rediscluster-datastore_rediscluster-{py37,py311,py312,pypy310}-redislatest,
-    solr-datastore_pysolr-{py27,py37,py38,py39,py310,py311,py312,pypy27,pypy310},
+    solr-datastore_pysolr-{py37,py38,py39,py310,py311,py312,pypy310},
 
 [testenv]
 deps =
     # Base Dependencies
     {py38,py39,py310,py311,py312,pypy310}: pytest==8.2.1
```

Comment:
We keep pytest pinned to prevent failures from updating it too often. We've had that in the past a lot. No one has updated it in a while. I'll open a tiny PR separately at some point to upgrade pytest and Webtest to the newest versions.